README.md

```diff
@@ -75,6 +75,7 @@ rustls does not and will not support:
 * Compression.
 * Discrete-log Diffie-Hellman.
 * Automatic protocol version downgrade.
+* Self-signed certificates (except with dangerous configuration enabled)
```

Comment:
```suggestion
* Self-signed certificates
```

I don't think it's accurate to say that Rustls supports self-signed certificates with the dangerous config option. 

The BoGo test you pointed to is passing `--insecure` to `tlsclient-mio`, and that ultimately ends up **disabling** certificate validation entirely (:warning:!): https://github.com/rustls/rustls/blob/460934b1b06670e8e84c3e99beadb6dfd8eee21c/examples/src/bin/tlsclient-mio.rs#L357-L363

In particular this means we're not:
* Checking that the self-signed certificate's signature is valid
* Checking that the self-signed certificate isn't expired
* Checking that the self-signed certificate has a subject name matching the name that was connected to

The mechanism that `tlsclient-mio` uses for the above is also an implementation detail specific to the example (a custom `ServerCertVerifier` impl), and not something you can get out-of-box with the Rustls library.

That's a lot of words (sorry!) but I think the right change here is to list self signed certificates as unsupported without further clarification. Does that make sense?